lib/membrane_live/hls/file_storage.ex

```diff
@@ -57,49 +60,63 @@ defmodule MembraneLive.HLS.FileStorage do
         contents,
         _metadata,
         %{mode: :binary},
-        %__MODULE__{directory: directory}
+        %{directory: directory} = state
       ) do
-    File.write(Path.join(directory, segment_filename), contents, [:binary])
+    {File.write(Path.join(directory, segment_filename), contents, [:binary]), state}
   end
 
   @impl true
-  def store(_parent_id, name, contents, _metadata, %{mode: :text}, %__MODULE__{
-        directory: directory
-      }) do
+  def store(
+        _parent_id,
+        name,
+        contents,
+        _metadata,
+        %{mode: :text},
+        %{directory: directory} = state
+      ) do
     result = File.write(Path.join(directory, name), contents)
 
-    if String.match?(name, ~r/\.m3u8$/) do
-      notify_playlist_update(directory, name, contents)
-    end
+    state =
+      if String.match?(name, ~r/\.m3u8$/) do
+        notify_playlist_update(name, contents, state)
+      else
+        state
+      end
 
-    result
+    {result, state}
   end
 
   @impl true
-  def remove(_parent_id, name, _ctx, %__MODULE__{directory: location}) do
```

Comment:
I think that state should still be the type of `__MODULE__` and config should be changed to some other type, but I'm not sure if I'm correct.

Reply:
I wanted to make it analogous to how it works inside Membrane's element.
For example in `RTC.Engine.Endpoint.HLS` the `Endpoint.HLS` struct is used for passing configuration to the `handle_init/2`function.
Inside `handle_init` we add some other fields to this map, but it is no longer a struct.

Reply:
it can stay this way. But I don't think membrane elements are the best example of the general rule, because it is framework specific more than elixir specific. You can look for an example at [track module](https://github.com/membraneframework/membrane_http_adaptive_stream_plugin/blob/master/lib/membrane_http_adaptive_stream/manifest/track.ex).